services/pull/merge.go

```diff
@@ -245,7 +236,7 @@ func Merge(ctx context.Context, pr *issues_model.PullRequest, doer *user_model.U
 }
 
 // doMergeAndPush performs the merge operation without changing any pull information in database and pushes it up to the base repository
-func doMergeAndPush(ctx context.Context, pr *issues_model.PullRequest, doer *user_model.User, mergeStyle repo_model.MergeStyle, expectedHeadCommitID, message string) (string, error) {
+func doMergeAndPush(ctx context.Context, pr *issues_model.PullRequest, doer *user_model.User, mergeStyle repo_model.MergeStyle, expectedHeadCommitID, message string, isMerge bool) (string, error) {
```

Comment:
Why it is called `isMerge`? This function `doMergeXxx`, why isMerge could be false?

(I have read the code and I can answer the question, but, I do not think such naming and design is right)

Reply:
I think the original function name `doMergeAndPush` should be `doCommitAndPush`.

Reply:
But I can see this function just does "merge and push", see below: createTemporaryRepoForMerge, switch mergeStyle

Reply:
3e320f865f1c93e1ce646da74f5cacc6c404e3e1